src/sentry/sentry_metrics/querying/eap/mql_eap_bridge.py

```diff
@@ -0,0 +1,131 @@
+from collections.abc import Iterable, Sequence
+from datetime import datetime, timedelta
+
+import requests
+import snuba_sdk.mql.mql
+from django.conf import settings
+from google.protobuf.timestamp_pb2 import Timestamp as ProtobufTimestamp
+from snuba_sdk import Timeseries
+from snuba_sdk.conditions import And as MQLAnd
+from snuba_sdk.conditions import Condition as MQLCondition
+from snuba_sdk.conditions import ConditionGroup
+from snuba_sdk.conditions import Op as MQLOp
+from snuba_sdk.conditions import Or as MQLOr
+
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.sentry_metrics.querying.eap.protobufs.AggregateBucket_pb2 import (
+    AggregateBucketRequest,
+    AggregateBucketResponse,
+)
+from sentry.sentry_metrics.querying.eap.protobufs.BaseRequest_pb2 import RequestInfo
+from sentry.sentry_metrics.querying.eap.protobufs.Filters_pb2 import (
+    AndFilter,
+    OrFilter,
+    StringFilter,
+    TraceItemFilter,
+)
+
+
+def parse_mql_filters(group: ConditionGroup) -> Iterable[TraceItemFilter]:
+    for cond in group:
+        if isinstance(cond, MQLAnd):
+            yield TraceItemFilter(
+                and_filter=AndFilter(filters=list(parse_mql_filters(cond.conditions)))
+            )
+        elif isinstance(cond, MQLOr):
+            yield TraceItemFilter(
+                and_filter=OrFilter(filters=list(parse_mql_filters(cond.conditions)))
+            )
+        elif isinstance(cond, MQLCondition):
+            if cond.op == MQLOp.EQ:
+                yield TraceItemFilter(
+                    string_comparison=StringFilter(key=cond.lhs.name, value=cond.rhs)
+                )
+        # TODO: maybe we want to implement other stuff
+
+
+def make_eap_request(
+    query_mql: str,
+    start: datetime,
+    end: datetime,
+    interval: int,
+    organization: Organization,
+    projects: Sequence[Project],
+    referrer: str,
+) -> dict:
+    start_time_proto = ProtobufTimestamp()
+    start_time_proto.FromDatetime(start)
+    end_time_proto = ProtobufTimestamp()
+    end_time_proto.FromDatetime(end)
+
+    ts: Timeseries = snuba_sdk.mql.mql.parse_mql(query_mql)
+
+    aggregate_map = {
+        "sum": AggregateBucketRequest.SUM,
+        "avg": AggregateBucketRequest.AVG,
+        "p50": AggregateBucketRequest.P50,
+        "p95": AggregateBucketRequest.P95,
+        "P99": AggregateBucketRequest.P99,
+        "count": AggregateBucketRequest.COUNT,
+    }
+
+    rpc_filters = None
+    if ts.filters is not None:
+        rpc_filters = TraceItemFilter(
+            and_filter=AndFilter(filters=list(parse_mql_filters(ts.filters)))
+        )
+    req = AggregateBucketRequest(
```

Comment:
How is this determining what the interval buckets should be when building the timeseries?

Reply:
hard coded for now